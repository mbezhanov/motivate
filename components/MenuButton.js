import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import { AntDesign, Octicons } from '@expo/vector-icons';
import StyledText from './StyledText';
import Sizes from "../constants/Sizes";
import Colors from '../constants/Colors';

const button = (props) => {
  const Icon = ['upload', 'download'].indexOf(props.icon) !== -1 ? AntDesign : Octicons;

  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <StyledText style={{ marginHorizontal: Sizes.S, marginVertical: Sizes.S / 2 }}>{props.label}</StyledText>
      </View>
      <TouchableOpacity style={styles.button} onPress={props.onPress}>
        <Icon name={props.icon} size={Sizes.M} color={Colors.HIGHLIGHT} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-end',
    marginBottom: Sizes.L / 2,
    flexDirection: 'row',
  },
  button: {
    width: Sizes.L * 2,
    height: Sizes.L * 2,
    borderRadius: Sizes.L,
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND,
  },
  textContainer: {
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND_TRANSPARENT,
    borderRadius: Sizes.S / 2,
    marginRight: Sizes.S / 2,
  },
});

export default button;
