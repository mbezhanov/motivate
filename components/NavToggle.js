import React from "react";
import {Animated, StyleSheet, TouchableWithoutFeedback, View} from 'react-native';
import { Octicons } from '@expo/vector-icons';
import Colors from '../constants/Colors';
import Sizes from '../constants/Sizes';

const navToggle = ({ onPress, rotate }) => (
  <TouchableWithoutFeedback onPress={onPress}>
    <View style={styles.button}>
      <Animated.View style={{ transform: [{ rotate }] }}>
        <Octicons name="plus" size={Sizes.M} color={Colors.HIGHLIGHT}  />
      </Animated.View>
    </View>
  </TouchableWithoutFeedback>
);

const styles = StyleSheet.create({
  button: {
    width: Sizes.XL * 2,
    height: Sizes.XL * 2,
    borderRadius: Sizes.XL,
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND,
  }
});

export default navToggle;
