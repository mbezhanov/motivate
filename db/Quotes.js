import { SQLite } from 'expo';

class Quotes {
  constructor() {
    this.db = SQLite.openDatabase('quotes.db');
    this.db.transaction(tx => {
      tx.executeSql('create table if not exists quotes (id integer primary key not null, content text, author text, book text, times_seen int default 0);');
    });
  }

  insert = (quote, author, book) => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('insert into quotes (content, author, book) values (?, ?, ?);', [quote, author, book], (tx, resultSet) => {
          resolve(resultSet.insertId);
        });
      }, error => reject(error));
    });

  };

  random = () => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('select * from quotes order by random() asc limit 1;', [], (tx, resultSet) => {
          const { rows: { _array }} = resultSet;
          const quote = _array.length > 0 ? _array[0] : null;

          if (quote) {
            quote.imageUrl = this._generateImageUrl(quote.content);
          }
          resolve(quote);
        });
      }, error => reject(error));
    });
  };

  _generateImageUrl = (quote) => {
    const MAX_ID = 1084;
    let sum = 0;

    for (let i = 0; i < quote.length; i++) {
      sum += quote.charCodeAt(i);
    }

    return `https://picsum.photos/480/960?image=${sum % MAX_ID}`;
  };
}

export default new Quotes();
