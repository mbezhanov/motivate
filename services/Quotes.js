import { SQLite } from 'expo';

class Quotes {
  constructor() {
    this.db = SQLite.openDatabase('quotes.db');
    this.db.transaction(tx => {
      tx.executeSql('create table if not exists quotes (id integer primary key not null, content text, author text, book text, times_seen int default 0);');
    });
  }

  insert = (quote, author, book) => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('insert into quotes (content, author, book) values (?, ?, ?);', [quote, author, book], (tx, resultSet) => {
          resolve(resultSet.insertId);
        });
      }, error => reject(error));
    });
  };

  update = (id, quote, author, book) => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('update quotes set content = ?, author = ?, book = ? where id = ?;', [quote, author, book, id], (tx, resultSet) => {
          resolve(resultSet.rowsAffected);
        });
      }, error => reject(error));
    });
  };

  delete = (id) => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('delete from quotes where id = ?;', [id], (tx, resultSet) => {
          resolve(resultSet.rowsAffected);
        });
      }, error => reject(error));
    });
  };

  random = () => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('select * from quotes order by random() asc limit 1;', [], (tx, resultSet) => {
          const { rows: { _array }} = resultSet;
          const quote = _array.length > 0 ? _array[0] : null;

          if (quote) {
            quote.imageUrl = this._generateImageUrl(quote.content);
          }
          resolve(quote);
        });
      }, error => reject(error));
    });
  };

  importCsv = (csv) => {
    return new Promise((resolve, reject) => {
      let rowsAffected = 0;
      this.db.transaction(tx => {
        csv.data.forEach(row => {
          const content = row[0];
          const author = row[1];
          const book = row[2];

          if (!content || !author || !book) {
            return;
          }
          tx.executeSql('insert into quotes (content, author, book, times_seen) values (?, ?, ?, ?);', [content, author, book, 0], (tx, resultSet) => {
            rowsAffected += resultSet.rowsAffected;
          });
        });
      }, error => reject(error), () => resolve(rowsAffected));
    });
  };

  exportCsv = () => {
    return new Promise((resolve, reject) => {
      this.db.transaction(tx => {
        tx.executeSql('select content, author, book, times_seen from quotes order by id asc;', [], (tx, resultSet) => {
          const { rows: { _array }} = resultSet;
          resolve(_array);
        });
      }, error => reject(error));
    });
  };

  _generateImageUrl = (quote) => {
    const MAX_ID = 1084;
    let sum = 0;

    for (let i = 0; i < quote.length; i++) {
      sum += quote.charCodeAt(i);
    }

    return `https://picsum.photos/480/960?image=${sum % MAX_ID}`;
  };
}

export default new Quotes();
