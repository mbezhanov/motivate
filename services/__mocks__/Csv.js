export default {
  importQuotes: jest.fn(),
  exportQuotes: jest.fn(),
};
