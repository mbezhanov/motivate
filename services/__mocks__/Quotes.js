export const IMPORT_MODE_APPEND = 1;
export const IMPORT_MODE_OVERWRITE = 2;

export default {
  importCsv: jest.fn(),
  exportCsv: jest.fn(),
};
