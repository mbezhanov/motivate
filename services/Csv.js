import { FileSystem, MailComposer } from 'expo';
import Papa from 'papaparse';
import moment from 'moment';
import Quotes from './Quotes';

const importQuotes = (documentUri, importMode) => {
  return FileSystem
    .readAsStringAsync(documentUri)
    .then(contents => new Promise((resolve, reject) => {
      const onImportFinished = count => resolve(count);
      const onImportFailed = () => reject('CSV import failed.');

      return Quotes.importCsv(Papa.parse(contents), importMode).then(onImportFinished).catch(onImportFailed);
    }));
};

const exportQuotes = () => {
  return Quotes
    .exportCsv()
    .then(quotes => {
      if (!Array.isArray(quotes) || quotes.length === 0) {
        throw new Error('document export failed');
      }
      const fileUri = `${FileSystem.documentDirectory}export-${moment().format('YYYY-MM-DD-HH-mm-ss')}.csv`;

      return FileSystem
        .writeAsStringAsync(fileUri, Papa.unparse(quotes))
        .then(() => {
          return MailComposer
            .composeAsync({
              subject: 'exported quotes',
              body: `Quotes exported on ${moment().format('YYYY-MM-DD-HH-mm-ss')}`,
              attachments: [fileUri]
            })
            .then(() => quotes.length);
        });
  });
};

export default { importQuotes, exportQuotes };
