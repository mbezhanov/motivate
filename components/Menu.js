import React, { Component } from 'react';
import { Alert, Animated, Easing, StyleSheet, View } from 'react-native';
import { DocumentPicker } from 'expo';
import MenuButton from './MenuButton';
import NavToggle from './NavToggle';
import Sizes from '../constants/Sizes';
import Colors from '../constants/Colors';
import Csv from '../services/Csv';
import { IMPORT_MODE_APPEND, IMPORT_MODE_OVERWRITE } from '../services/Quotes';
import PropTypes from 'prop-types';

const ANIMATION_DURATION = 200; // milliseconds;

export default class Menu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isAnimating: false,
      isExpanded: false,
      rotation: new Animated.Value(0),
      slideIn: new Animated.Value(0),
    }
  }

  render() {
    const right = this.state.slideIn.interpolate({
      inputRange: [0, 1],
      outputRange: [Sizes.XL * 5 * -1, 8],
    });
    const rotate = this.state.rotation.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '225deg'],
    });

    return (
      <View style={styles.container}>
        <Animated.View style={{ position: 'relative', right }}>
          <MenuButton key="import" label="Import" icon="download" onPress={this._onImportButtonPress}/>
          <MenuButton key="export" label="Export" icon="upload" onPress={this._onExportButtonPress}/>
          <MenuButton key="add" label="Add" icon="plus" onPress={this._onAddButtonPress}/>
          <MenuButton key="edit" label="Edit" icon="pencil" onPress={this._onEditButtonPress}/>
          <MenuButton key="delete" label="Delete" icon="trashcan" onPress={this._onDeleteButtonPress}/>
        </Animated.View>
        <NavToggle onPress={this._onPlusButtonPress} rotate={rotate}/>
      </View>
    );
  }

  _onImportButtonPress = () => {
    this._toggleMenuItems(() => {
      const onImportSuccess = (count) => {
        Alert.alert('Success', `Imported ${count} quotes.`, [
          { text: 'OK', onPress: () => this.props.onSuccessfulImport() },
        ]);
      };
      const onImportFail = () => {
        Alert.alert('Error', 'An error occurred.');
      };
      DocumentPicker
        .getDocumentAsync()
        .then(document => {

          if (document.type === 'cancel') {
            Alert.alert('Import canceled', '');
            return;
          }
          const shouldShowPrompt = (this.props.quote !== null);

          if (shouldShowPrompt) {
            Alert.alert('Import Settings', 'Would you like to append quotes to your existing collection or overwrite it completely?', [
              { text: 'Overwrite', onPress: () => Csv.importQuotes(document.uri, IMPORT_MODE_OVERWRITE).then(onImportSuccess).catch(onImportFail) },
              { text: 'Append', onPress: () => Csv.importQuotes(document.uri, IMPORT_MODE_APPEND).then(onImportSuccess).catch(onImportFail) }
            ]);
          } else {
            Csv.importQuotes(document.uri, IMPORT_MODE_APPEND).then(onImportSuccess).catch(onImportFail);
          }
        });
    });
  };

  _onExportButtonPress = () => {
    this._toggleMenuItems(() => {
      Csv
        .exportQuotes()
        .then(count => {
          Alert.alert('Success', `Exported ${count} quotes.`);
        })
        .catch(() => {
          Alert.alert('Error', 'An error occurred.');
        });
    });
  };

  _onAddButtonPress = () => {
    this._toggleMenuItems(() => {
      this.props.navigation.push('Form', {
        headerTitle: 'Add Quote',
      });
    });
  };

  _onEditButtonPress = () => {
    this._toggleMenuItems(() => {
      this.props.navigation.push('Form', {
        quote: this.props.quote,
        headerTitle: 'Edit Quote',
      });
    });
  };

  _onDeleteButtonPress = () => {
    this._toggleMenuItems(() => {
      Alert.alert('Delete Quote', 'Are you sure you want to delete this quote?', [
        { text: 'OK', onPress: () => this.props.onDelete() },
        { text: 'Cancel', style: 'cancel' },
      ]);
    })
  };

  _onPlusButtonPress = () => {
    this._toggleMenuItems();
  };

  _toggleMenuItems = (onAnimationEnd = () => {}) => {
    if (this.state.isAnimating) {
      return;
    }
    const toValue = 0 + !this.state.isExpanded;

    Animated.parallel([
      Animated.timing(this.state.rotation, {
        toValue,
        duration: ANIMATION_DURATION,
        easing: Easing.linear
      }),
      Animated.timing(this.state.slideIn, {
        toValue,
        duration: ANIMATION_DURATION,
        easing: Easing.linear,
      })
    ]).start(() => {
      this.setState({
        isAnimating: false,
        isExpanded: !this.state.isExpanded,
      }, () => {
        onAnimationEnd();
      });
    });
  };
}

Menu.propTypes = {
  quote: PropTypes.shape({
    id: PropTypes.number,
    author: PropTypes.string,
    book: PropTypes.string,
    content: PropTypes.string,
    imageUrl: PropTypes.string,
    times_seen: PropTypes.number,
  }),
  navigation: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
  onSuccessfulImport: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: Sizes.M,
    right: Sizes.M,
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
  },
  button: {
    width: Sizes.L * 2,
    height: Sizes.L * 2,
    borderRadius: Sizes.L,
    textAlign: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.BACKGROUND,
    marginBottom: Sizes.S,
  },
});
