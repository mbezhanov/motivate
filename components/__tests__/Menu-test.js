import React from 'react';
import { Alert } from 'react-native';
import * as DocumentPicker from 'expo-document-picker';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import Menu from '../Menu';
import MenuButton from '../MenuButton';
import NavToggle from '../NavToggle';
import Csv from '../../services/Csv';
import { IMPORT_MODE_APPEND, IMPORT_MODE_OVERWRITE } from '../../services/Quotes';

jest.mock('../MenuButton', () => 'MenuButton');
jest.mock('../NavToggle', () => 'NavToggle');

jest.mock('Animated', () => {
  const ActualAnimated = jest.requireActual('Animated');
  return {
    ...ActualAnimated,
    timing: (value, config) => {
      return { value, config };
    },
    parallel: (animations) => {
      return {
        start: (callback) => {
          for (let i in animations) {
            let animation = animations[i];
            animation.value.setValue(animation.config.toValue);
          }
          callback && callback();
        },
      };
    },
  };
});
jest.mock('expo', () => ({
  DocumentPicker: {
    getDocumentAsync: jest.fn(),
  },
}));
jest.mock('../../services/Csv');
jest.mock('../../services/Quotes');
jest.mock('Alert', () => ({ alert: jest.fn() }));

const _press = (component) => {
  component.props.onPress(); // simulate press on component
};

const _toggleSubmenu = (treeRoot) => {
  const navToggle = treeRoot.findByType(NavToggle);
  _press(navToggle);
};

const mockImportedDocument = {
  name: 'test.csv',
  size: 64,
  type: 'success',
  uri: 'file:///data/user/0/host.exp.exponent/cache/ExperienceData/%2540mbezhanov%252Fmotivate/DocumentPicker/9e018921-301a-4c0b-9869-f69db3ed760fcsv'
};

describe('Menu', () => {
  let tree;
  let navigation;
  const quote = { id: 123, content: 'foo', author: 'bar', book: 'baz' };
  const onDelete = jest.fn();
  const onSuccessfulImport = jest.fn();

  beforeEach(() => {
    navigation = { push: jest.fn() };
    tree = renderer.create(
      <Menu
        navigation={navigation}
        quote={quote}
        onDelete={onDelete}
        onSuccessfulImport={onSuccessfulImport}
      />
    );
    Csv.importQuotes.mockClear();
    Alert.alert.mockClear();
  });

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('displays all hidden submenu items when the nav toggle switch is pressed', () => {
    const a = tree.toJSON();
    const navToggle = tree.root.findByType(NavToggle);
    _press(navToggle);
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });

  it('hides all displayed submenu items when the nav toggle switch is pressed', () => {
    const navToggle = tree.root.findByType(NavToggle);
    _press(navToggle); // open submenu items
    const a = tree.toJSON();
    _press(navToggle); // close submenu items
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });

  it('lets the user pick a document when an import is requested', async () => {
    DocumentPicker.getDocumentAsync.mockResolvedValueOnce(mockImportedDocument);
    Csv.importQuotes.mockResolvedValueOnce(123);
    _toggleSubmenu(tree.root);
    const a = tree.toJSON();
    const importButton = tree.root.findByProps({ label: 'Import' });
    await _press(importButton);
    expect(DocumentPicker.getDocumentAsync).toBeCalledTimes(1);
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(diff).toMatchSnapshot();
  });

  it('directly delegates control to the CSV service if the database is empty', async () => {
    tree = renderer.create(
      <Menu
        navigation={navigation}
        quote={null}
        onDelete={onDelete}
        onSuccessfulImport={onSuccessfulImport}
      />
    );
    DocumentPicker.getDocumentAsync.mockResolvedValueOnce(mockImportedDocument);
    Csv.importQuotes.mockResolvedValueOnce(123);
    _toggleSubmenu(tree.root);
    const importButton = tree.root.findByProps({ label: 'Import' });
    await _press(importButton);
    expect(Csv.importQuotes).toBeCalledTimes(1);
  });

  it('asks the user whether new quotes should be appended to or overwrite the existing collection, if the database is not empty', async () => {
    _toggleSubmenu(tree.root);
    DocumentPicker.getDocumentAsync.mockResolvedValueOnce(mockImportedDocument);
    Csv.importQuotes.mockResolvedValueOnce(123);
    const importButton = tree.root.findByProps({ label: 'Import' });
    await _press(importButton);
    expect(Alert.alert).toBeCalledTimes(1);
    expect(Alert.alert.mock.calls[0][2][0].text).toEqual('Overwrite');
    expect(Alert.alert.mock.calls[0][2][1].text).toEqual('Append');
  });

  it('can instruct the Csv service to overwrite the existing collection', async () => {
    _toggleSubmenu(tree.root);
    DocumentPicker.getDocumentAsync.mockResolvedValueOnce(mockImportedDocument);
    Csv.importQuotes.mockResolvedValueOnce(123);
    const importButton = tree.root.findByProps({ label: 'Import' });
    await _press(importButton);
    Alert.alert.mock.calls[0][2][0].onPress();
    expect(Csv.importQuotes).toBeCalledTimes(1);
    expect(Csv.importQuotes.mock.calls[0][0]).toEqual(mockImportedDocument.uri);
    expect(Csv.importQuotes.mock.calls[0][1]).toEqual(IMPORT_MODE_OVERWRITE);
  });

  it('can instruct the Csv service to append new quotes to the existing collection', async () => {
    _toggleSubmenu(tree.root);
    DocumentPicker.getDocumentAsync.mockResolvedValueOnce(mockImportedDocument);
    Csv.importQuotes.mockResolvedValueOnce(123);
    const importButton = tree.root.findByProps({ label: 'Import' });
    await _press(importButton);
    Alert.alert.mock.calls[0][2][1].onPress();
    expect(Csv.importQuotes).toBeCalledTimes(1);
    expect(Csv.importQuotes.mock.calls[0][0]).toEqual(mockImportedDocument.uri);
    expect(Csv.importQuotes.mock.calls[0][1]).toEqual(IMPORT_MODE_APPEND);
  });

  it('delegates control to the CSV service when an export is requested', async () => {
    Csv.exportQuotes.mockResolvedValueOnce(123);
    const a = tree.toJSON();
    const exportButton = tree.root.findByProps({ label: 'Export' });
    await _press(exportButton);
    const b = tree.toJSON();
    const diff = snapshotDiff(a, b);
    expect(Alert.alert).toBeCalledTimes(1);
    expect(Alert.alert.mock.calls[0][0]).toEqual('Success');
    expect(Alert.alert.mock.calls[0][1].indexOf('123')).not.toEqual(-1);
    expect(diff).toMatchSnapshot();
  });

  it('displays an error message if the CSV export fails', async () => {
    Csv.exportQuotes.mockRejectedValueOnce();
    const exportButton = tree.root.findByProps({ label: 'Export' });
    await _press(exportButton);
    expect(Alert.alert).toBeCalledTimes(1);
    expect(Alert.alert.mock.calls[0][0]).toEqual('Error');
  });

  it('navigates to the Form screen when the insertion of a new quote is requested', () => {
    const addButton = tree.root.findByProps({ label: 'Add' });
    _press(addButton);
    expect(navigation.push).toBeCalledTimes(1);
    expect(navigation.push.mock.calls[0][0]).toEqual('Form');
  });

  it('navigates to the Form screen when the update of an existing quote is requested', () => {
    const editButton = tree.root.findByProps({ label: 'Edit' });
    _press(editButton);
    expect(navigation.push).toBeCalledTimes(1);
    expect(navigation.push.mock.calls[0][0]).toEqual('Form');
    expect(navigation.push.mock.calls[0][1].quote).toEqual(quote);
  });

  it('asks the user for confirming the intention of deleting a particular quote', () => {
    const deleteButton = tree.root.findByProps({ label: 'Delete' });
    _press(deleteButton);
    expect(Alert.alert).toBeCalledTimes(1);
    expect(Alert.alert.mock.calls[0][0].indexOf('Delete')).not.toEqual(-1);
  });

  it('raises an event if the user confirms the intention of deleting a particular quote', () => {
    const deleteButton = tree.root.findByProps({ label: 'Delete' });
    _press(deleteButton);
    expect(Alert.alert).toBeCalledTimes(1);
    const okButton = Alert.alert.mock.calls[0][2][0];
    okButton.onPress();
    expect(onDelete).toBeCalledTimes(1);
  });
});
