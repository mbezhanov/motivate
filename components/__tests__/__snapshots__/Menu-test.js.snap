// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu delegates control to the CSV service when an export is requested 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,11 @@
  >
    <View
      style={
        Object {
          \\"position\\": \\"relative\\",
-         \\"right\\": -140,
+         \\"right\\": 8,
        }
      }
    >
      <MenuButton
        icon=\\"download\\""
`;

exports[`Menu displays all hidden submenu items when the nav toggle switch is pressed 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,11 @@
  >
    <View
      style={
        Object {
          \\"position\\": \\"relative\\",
-         \\"right\\": -140,
+         \\"right\\": 8,
        }
      }
    >
      <MenuButton
        icon=\\"download\\""
`;

exports[`Menu hides all displayed submenu items when the nav toggle switch is pressed 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,11 @@
  >
    <View
      style={
        Object {
          \\"position\\": \\"relative\\",
-         \\"right\\": 8,
+         \\"right\\": -140,
        }
      }
    >
      <MenuButton
        icon=\\"download\\""
`;

exports[`Menu lets the user pick a document when an import is requested 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,11 +11,11 @@
  >
    <View
      style={
        Object {
          \\"position\\": \\"relative\\",
-         \\"right\\": 8,
+         \\"right\\": -140,
        }
      }
    >
      <MenuButton
        icon=\\"download\\""
`;

exports[`Menu renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "bottom": 16,
      "justifyContent": "flex-end",
      "position": "absolute",
      "right": 16,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
        "right": -140,
      }
    }
  >
    <MenuButton
      icon="download"
      label="Import"
      onPress={[Function]}
    />
    <MenuButton
      icon="upload"
      label="Export"
      onPress={[Function]}
    />
    <MenuButton
      icon="plus"
      label="Add"
      onPress={[Function]}
    />
    <MenuButton
      icon="pencil"
      label="Edit"
      onPress={[Function]}
    />
    <MenuButton
      icon="trashcan"
      label="Delete"
      onPress={[Function]}
    />
  </View>
  <NavToggle
    onPress={[Function]}
    rotate={"0deg"}
  />
</View>
`;
